### PR TITLE
tests:Migrate docker container to Ubuntu22.04 for integration test cases.

### PR DESCRIPTION
**Pull request by peng6662001**
Integration test's docker image is currently based on 20.04. This commit migrates the docker image to 22.04.
All integration test cases pass on Ampere Altra.

### CODE REVIEW

**Comment by rbradford** [comment]
Thank you for looking into this - please respect the commit message style of the project. A great way to do that is to run `git log` over the file(s) your changing to understand how they should be formatted. In particular your use of the word "image" makes it look like you are talking about the disk image used for the tests - but this is the development container.

**Comment by peng6662001** [comment]
> Thank you for looking into this - please respect the commit message style of the project. A great way to do that is to run `git log` over the file(s) your changing to understand how they should be formatted. In particular your use of the word "image" makes it look like you are talking about the disk image used for the tests - but this is the development container.

Hi @rbradford ,I re-edited the comment, any more questions?

File: scripts/run_integration_tests_aarch64.sh [hunk 1]
```diff
@@ -30,13 +30,14 @@ build_spdk_nvme() {
...
     cp -r "$WORKLOADS_DIR/spdk/scripts/rpc" $SPDK_DEPLOY_DIR/rpc
+    cp -r "$WORKLOADS_DIR/spdk/python" $SPDK_DEPLOY_DIR/../
 }
 
 build_virtiofsd() {
     VIRTIOFSD_DIR="$WORKLOADS_DIR/virtiofsd_build"
     VIRTIOFSD_REPO="https://gitlab.com/virtio-fs/virtiofsd.git"
 
-    checkout_repo "$VIRTIOFSD_DIR" "$VIRTIOFSD_REPO" v1.1.0 "220405d7a2606c92636d31992b5cb3036a41047b"
+    checkout_repo "$VIRTIOFSD_DIR" "$VIRTIOFSD_REPO" v1.6.1 "7886ab7c819fd8de370d5a3bbfaae44de56923ed"
```
[the remark this answers is not in the document]
Comment (peng6662001) [reply scripts/run_integration_tests_aarch64.sh:40]: Hi @rbradford ,I have updated the patch and the comment,is there other problem?

**Comment by michael2012z** [comment]
The integration test failed. Seemingly you are seeing the same problem as in this PR: https://github.com/cloud-hypervisor/cloud-hypervisor/pull/5072#issuecomment-1376341417

File: resources/Dockerfile [hunk 2]
```diff
@@ -40,7 +40,7 @@ RUN apt-get update \
 	socat \
 	dosfstools \
 	cpio \
-	python \
+	gcc-9 \
```
[the remark this answers is not in the document]
Comment (peng6662001) [reply resources/Dockerfile]: @rbradford I've removed `gcc-9`, but compilation still fails. I will continue to research it.

[the remark this answers is not in the document]
Comment (peng6662001) [reply resources/Dockerfile]: @rbradford @michael2012z 
```
/cargo/registry/src/github.com-1ecc6299db9ec823/compiler_builtins-0.1.82/./lib/builtins/cpu_model.c:829: undefined reference to `getauxval'
```
I have compiled multi versions of musl-gcc,but can't fix this error, could you please give me some other suggestions?

**Review by likebreath — COMMENTED** [review]
Can you please breakdown the PR into multiple commits where each commit focuses on one thing and self-explainable (with commit message)? 

I see the PR touches quite many things more than bumping base docker image, say use new versions of SPDK and VIRTIOFSD, override default gcc for compilation, rebuild a custom version of musl-tools, etc. It would great if you can have separate commits for each of them with a commit message of "what and why".

Also, I can help with debugging the ovs_dpdk live-migration failures when this is the last blocking issue.

**Comment by michael2012z** [comment]
@peng6662001 A new commit was added to your branch. I was trying to resolve a conflict to unblock the CI. Please remove that commit when you rebase your branch.

**Pull request closed without merging**